website/docs/developer/research/research-reconciliation.mdx

```diff
@@ -0,0 +1,683 @@
+  being used in some ORCA code.
+
+
+## Comparison between ORCA S3 and the ORCA Catalog
+
+In addition to the comparison between the Cumulus catalog and ORCA, ORCA will
+also need to manage and maintain the catalog holdings between the ORCA database
+and the actual contents of the ORCA S3 buckets. This will require comparing the
+*Key*, *LastModified*, *ETag*, and *Size* keys to the of the files in the bucket
+to the ORCA holdings catalog. Discrepancies should be reported back to the user
+and detail orphan and missing files. We could possibly create a temporary reporting
+table to hold these values that the Cumulus Dashboard could display for us.
+
+Some things to think through when designing this solution are factors of scale
+and managing them. To provide some thought and guidance, LP DAAC one of the larger
+DAACs has approximately 300 million files in the archive and growing. If all the
+files are also stored in ORCA, we will need to reconcile all the files in a smart
+and somewhat quick way on a semi regular basis.
+
+Ideally, this reconciliation would occur on a regular schedule using a rule and
+reports would be stored in the database for users to retrieve the information.
+Additionally, the code should handle failures gracefully and be able to pick up
+where it left off either gathering the current contents in S3 or performing the
+ORCA catalog internal comparisons with S3.
+
+Future enhancements may include displaying the information on the Cumulus dashboard,
+notifying operators via email or other means of discrepancies, or automated healing
+of the archive based on reconciliation results.
+
+
+### Getting a Listing from AWS S3
+
+To get a file listing of an S3 bucket, the [list_objects_v2 method](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/s3.html#S3.Client.list_objects_v2)
+can be used from the boto3 library. An example of the useage can be seen below.
+Some notable items if this library is used includes the following.
+- Only a max of 1000 objects are returned per call. Additional logic will be needed to page through and get all of the results.
+- The listing can be filtered by prefix. Possibly some room here to add configuration to look at the collection lever if a collection can be denoted by prefix.
+- Version information is not included, only the latest file information is provided.
+- An additional call will need to be made per object to look at versions available for a file.
+- Only limited fields are returned from the call.
+- Each list call does cost in AWS (tiny fractions of a cent).
+
+Other considerations. It may be easier to load the data from the listing calls
+into a temporary table in the database and perform a handful of queries to find
+the delta information instead of looping through and doing a comparison in memory.
+
+#### Example Using list_objects_v2
+
+```python
+import boto3
+
+# Set the variables
+s3_client = boto3.resource('s3')
```

Comment:
should this be boto3.client('s3') since it's saying create the client?

Reply:
Either could be used. See the actual call that occurs later `s3_client.meta.client.` This is really just sample code to show the basics of the call and the outputs to provide context for further design.